agent/azure-monitor-exporter/src/main/java/com/azure/monitor/opentelemetry/exporter/implementation/pipeline/TelemetryItemExporter.java

```diff
@@ -113,9 +146,16 @@ public CompletableResultCode shutdown() {
     return listener.shutdown();
   }
 
-  CompletableResultCode internalSendByConnectionString(
+  CompletableResultCode internalSendByConnectionStringAndRoleName(
       List<TelemetryItem> telemetryItems, String connectionString) {
     List<ByteBuffer> byteBuffers;
+
+    // Don't send _OTELRESOURCE_ custom metric when OTEL_RESOURCE_ATTRIBUTES env var is empty
+    // insert _OTELRESOURCE_ at the beginning of each batch
+    if (!attributes.isEmpty()) {
+      telemetryItems.add(
+          0, createOtelResourceMetric(connectionString, telemetryItems.get(0).getTags()));
```

Comment:
I think would be slightly clearer here to pass `attributes` to the method here (since we just made decision above based on its presence), and could then make `createOtelResourceMetric` method static